python/mlx/nn/losses.py

```diff
@@ -2,7 +2,46 @@
 
 import mlx.core as mx
 
+def cross_entropy(logits: mx.array, targets: mx.array, axis: int = -1, reduction: str = 'mean'):
+    """
+    Computes the cross entropy loss between logits and targets.
+
+    Args:
+        logits (mx.array): The predicted logits.
+        targets (mx.array): The target values.
+        axis (int, optional): The axis over which to compute softmax. Defaults to -1.
+        reduction (str, optional): Specifies the reduction to apply to the output: 'none' | 'mean' | 'sum'. 
+                                    'none': no reduction will be applied. 
+                                    'mean': the sum of the output will be divided by the number of elements in the output.
+                                    'sum': the output will be summed. 
+                                    Defaults to 'mean'.
+
+    Returns:
+        mx.array: The computed cross entropy loss.
+    """
+    logits = logits - mx.max(logits, axis=axis, keepdims=True)
+    log_probs = mx.log(mx.softmax(logits, axis=axis))
+    score = mx.take_along_axis(log_probs, targets[..., None], axis).squeeze(-1)
```

Comment:
Would you mind reverting this to the old implementation, it is faster and numerically better as it avoids the explicit `softmax`. Also FYI for future reference `mx.softmax` is numerically stable you don't need to subtract the max for it.